operator/webserver.py

```diff
@@ -41,12 +67,50 @@ def mission_stop():
             # wait for it to finish
             interfaceprocess.wait()
 
-    return {'message': 'Mission stopped'}
+    car_controls.brake = 1
+    client.setCarControls(car_controls)
+    car_controls.brake = 0 #remove brake
+
+    log = '{}: {}'.format(str(datetime.now()), 'Mission stopped.')
+    logs.append(log)
+    del logs[:]
+
+    log_file.write(log + '\n')
+    log_file.close()
+
+    return {'response': log}
 
 @app.route('/mission/reset', methods=['POST'])
 def mission_reset():
     client.reset()
-    return {'message': 'Car reset'}
+
+    log = '{}: {}'.format(str(datetime.now()), 'Car reset.')
+    logs.append(log)
+    log_file.write(log + '\n')
+
+    return {'response': log}
+
+@app.route('/logs', methods=['GET'])
+def get_logs():
+    return {'response': logs}
+
+def referee_state_listener():
+    Timer(2.0, referee_state_listener).start()
+    ref = client.getRefereeState()
+    global doo_count, laps
+
+    if doo_count != ref.doo_counter:
+        doo_count = ref.doo_counter
```

Comment:
Sooo can we check for 2 cones hit right after each other?

like here it is missing 1 and 2.
![Screenshot from 2020-05-24 01-26-40](https://user-images.githubusercontent.com/8008227/82742442-78f48980-9d5e-11ea-8e97-09f6fec3ecc3.png)

Like in this case we can detect that we jumped from 0 to 2 cones, so we should print an extra message for the first hit cone.

To make it better to understand, can we change the message to something like `10th cone hit, total DOO is now 10`?

Reply:
I think this is caused by the server checking the referee state only once every 2 seconds. Do we know how many times per second the referee state gets refreshed?

Reply:
Fixed. referee state is checked twice every second now. Plus added to logic you mentioned: 'Like in this case we can detect that we jumped from 0 to 2 cones, so we should print an extra message for the first hit cone.'